src/query_param.c

```diff
@@ -159,7 +159,15 @@ void QueryParam_InitParams(QueryParam *p, size_t num) {
   memset(p->params, 0, sizeof(*p->params) * num);
 }
 
-int QueryParam_Resolve(Param *param, dict *params, QueryError *status) {
+/**
+ * Checks if the given numeric value is not empty.
+ * If the dialect version is 2 or higher, the value must not be empty, otherwise it can be 0 for backward compatibility.
+ */
+static inline bool checkNumericValueValid(const char *val, unsigned int dialectVersion) {
```

Comment:
Since this validates geo as well, consider expressing that in the function name.
Regardless, add it to the comment.